cookbooks/aws-parallelcluster-slurm/recipes/update_shared_storages.rb

```diff
@@ -0,0 +1,165 @@
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance with the
+# License. A copy of the License is located at
+#
+# http://aws.amazon.com/apache2.0/
+#
+# or in the "LICENSE.txt" file accompanying this file. This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES
+# OR CONDITIONS OF ANY KIND, express or implied. See the License for the specific language governing permissions and
+# limitations under the License.
+
+load_shared_storages_mapping
+
+ruby_block "get storage to mount and unmount" do
+  block do
+    # get ebs to unmount
+    node.default['cluster']['unmount_shared_dir_array'], node.default['cluster']['unmount_vol_array'] = get_ebs("unmount")
+    # get ebs to mount
+    node.default['cluster']['mount_shared_dir_array'], node.default['cluster']['mount_vol_array'] = get_ebs("mount")
+    # get efs to unmount
+    node.default['cluster']['unmount_efs_shared_dir_array'], node.default['cluster']['unmount_efs_fs_id_array'] = get_efs("unmount")
+    # get efs to mount
+    node.default['cluster']['mount_efs_shared_dir_array'], node.default['cluster']['mount_efs_fs_id_array'] = get_efs("mount")
+    # get fsx to unmount
+    node.default['cluster']['unmount_fsx_fs_id_array'], node.default['cluster']['unmount_fsx_fs_type_array'], node.default['cluster']['unmount_fsx_shared_dir_array'], node.default['cluster']['unmount_fsx_dns_name_array'], node.default['cluster']['unmount_fsx_mount_name_array'], node.default['cluster']['unmount_fsx_volume_junction_path_array'] = get_fsx("unmount")
+    # get fsx to mount
+    node.default['cluster']['mount_fsx_fs_id_array'], node.default['cluster']['mount_fsx_fs_type_array'], node.default['cluster']['mount_fsx_shared_dir_array'], node.default['cluster']['mount_fsx_dns_name_array'], node.default['cluster']['mount_fsx_mount_name_array'], node.default['cluster']['mount_fsx_volume_junction_path_array'] = get_fsx("mount")
+  end
+
+  def get_ebs(action)
+    # get ebs to mount or unmount
+    if action == "unmount"
+      in_shared_storages_mapping = node['cluster']['shared_storages_mapping']
+      not_in_shared_storages_mapping = node['cluster']['update_shared_storages_mapping']
+    elsif action == "mount"
+      in_shared_storages_mapping = node['cluster']['update_shared_storages_mapping']
+      not_in_shared_storages_mapping = node['cluster']['shared_storages_mapping']
+    end
+    shared_dir_array = []
+    vol_array = []
+    unless in_shared_storages_mapping["ebs"].nil?
+      in_shared_storages_mapping["ebs"].each do |storage|
+        next unless not_in_shared_storages_mapping["ebs"].nil? || !not_in_shared_storages_mapping["ebs"].include?(storage)
+        shared_dir_array.push(storage["mount_dir"])
+        vol_array.push(storage["volume_id"])
+      end
+    end
+    [shared_dir_array, vol_array]
+  end
+
+  def get_efs(action)
+    # get efs to mount or unmount
+    if action == "unmount"
```

Comment:
We use the action strings "mount" and "unmount" in more places. What about having constants for them?

Reply:
Currently we don't have somewhere to store constant in cookbook.

Reply:
I'm not getting this point. Why can't we do as follows in this file?

```
MOUNT_ACTION = "mount"
UNMOUNT_ACTION = "unmount"

....


if action == MOUNT_ACTION:
    do stuff
if action == UNMOUNT_ACTION:
    do stuff
```

Reply:
Oh you mean put the constant at the beginning of this file instead of a dedicated file. Yes we can do that, changed